Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/checkins/CheckInsViewModel.kt

```diff
@@ -0,0 +1,68 @@
+package de.rki.coronawarnapp.ui.eventregistration.attendee.checkins
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.SavedStateHandle
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.eventregistration.checkins.qrcode.QRCodeUriParser
+import de.rki.coronawarnapp.eventregistration.checkins.qrcode.QRCodeVerifier
+import de.rki.coronawarnapp.eventregistration.checkins.qrcode.QRCodeVerifyResult
+import de.rki.coronawarnapp.exception.ExceptionCategory
+import de.rki.coronawarnapp.exception.reporting.report
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
+import timber.log.Timber
+
+class CheckInsViewModel @AssistedInject constructor(
+    @Assisted private val savedState: SavedStateHandle,
+    @Assisted private val deepLink: String?,
+    dispatcherProvider: DispatcherProvider,
+    private val qrCodeVerifier: QRCodeVerifier,
+    private val qrCodeUriParser: QRCodeUriParser
+) : CWAViewModel(dispatcherProvider) {
+
+    private val verifyResultData = MutableLiveData<QRCodeVerifyResult>()
+    val verifyResult: LiveData<QRCodeVerifyResult> = verifyResultData
+
+    init {
```

Comment:
Will this still work in case the activity is open on this Screen? and when it is coming from Scanning fragment?
1- Case 1: there could be an uri coming from `newIntent` in main activity 
2- Case 2: when navigating to scan fragment and then back here again
is the view model created newly that is why is added in `init` ?

Reply:
It worked in my tests :thinking:, we are further a long we will test this again too.

> 1- Case 1: there could be an uri coming from newIntent in main activity

Causes navigation via navgraph. Navargs use fragment arguments under the hood, fragment arguments can only be set on new fragments. So a new fragment is created, and this did create a new viewmodel in my tests as the fragment viewmodels are scoped to the fragment.
This would would become an issue if we scope the viewmodel to an activity.

> 2- Case 2: when navigating to scan fragment and then back here again
is the view model created newly that is why is added in init ?

It would depend on if the checkins fragment is destroyed in that time. If it is not destroyed, nothing happens because `init` is not called again. If it is destroyed (low memory / don't keep activies), but because it was still on the backstack, we will get a `SavedStateHandle` with `SKEY_LAST_DEEPLINK` and then know not to execute the deepLink action again.